lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.Model.rb

```diff
@@ -914,6 +914,225 @@ def model_set_central_preheat_coil_spm(model, thermalZones, coil)
+         thermal_zone.additionalProperties.getFeatureAsBoolean('airloop dcv required by 901').get &&
+         !thermal_zone.additionalProperties.getFeatureAsBoolean('zone DCV implemented in user model').get
+        OpenStudio.logFree(OpenStudio::Error, 'openstudio.standards.Model', "For thermal zone #{thermal_zone.name}, ASHRAE 90.1 2019 6.4.3.8 requires this zone to have demand control ventilation, but it was not implemented in the user model, Appendix G baseline generation should be terminated!")
+      end
+    end
+  end
+
+  # Check if zones in the baseline model (to be created) should have DCV based on 90.1 2019 G3.1.2.5. Zone additional
+  # property 'apxg no need to have DCV' added
+  #
+  # @author Xuechen (Jerry) Lei, PNNL
+  # @param model [OpenStudio::Model::Model] Openstudio model
+  def model_add_apxg_dcv_properties(model)
+    model.getAirLoopHVACs.each do |air_loop_hvac|
+      if air_loop_hvac.airLoopHVACOutdoorAirSystem.is_initialized
+        oa_flow_m3_per_s = get_airloop_hvac_design_oa_from_sql(air_loop_hvac)
+      else
+        OpenStudio.logFree(OpenStudio::Info, 'openstudio.standards.AirLoopHVAC', "For #{air_loop_hvac.name}, DCV not applicable because it has no OA intake.")
+        return false
+      end
+      oa_flow_cfm = OpenStudio.convert(oa_flow_m3_per_s, 'm^3/s', 'cfm').get
+      if oa_flow_cfm <= 3000
+        air_loop_hvac.thermalZones.each do |thermal_zone|
+          thermal_zone.additionalProperties.setFeature('apxg no need to have DCV', true)
+        end
+      else # oa_flow_cfg > 3000, check zone people density
+        air_loop_hvac.thermalZones.each do |thermal_zone|
+          area_served_m2 = 0
+          num_people = 0
+          thermal_zone.spaces.each do |space|
+            area_served_m2 += space.floorArea
+            num_people += space.numberOfPeople
+          end
+          area_served_ft2 = OpenStudio.convert(area_served_m2, 'm^2', 'ft^2').get
+          occ_per_1000_ft2 = num_people / area_served_ft2 * 1000
+          if occ_per_1000_ft2 <= 100
+            thermal_zone.additionalProperties.setFeature('apxg no need to have DCV', true)
+          else
+            thermal_zone.additionalProperties.setFeature('apxg no need to have DCV', false)
+          end
+        end
+      end
+    end
+    # if a zone does not have this additional property, it means it was not served by airloop.
+  end
+
+  # Set DCV in baseline HVAC system if required
+  #
+  # @author Xuechen (Jerry) Lei, PNNL
+  # @param model [OpenStudio::Model::Model] Openstudio model
+  def model_set_baseline_demand_control_ventilation(model, climate_zone)
+    model.getAirLoopHVACs.each do |air_loop_hvac|
+      if baseline_air_loop_hvac_demand_control_ventilation_required?(air_loop_hvac)
+        air_loop_hvac_enable_demand_control_ventilation(air_loop_hvac, climate_zone)
+        air_loop_hvac.thermalZones.sort.each do |zone|
+          unless baseline_thermal_zone_demand_control_ventilation_required?(zone)
```

Comment:
Also need to account for situation where user model does not have DCV, but it is required in the Proposed and Baseline models. If the user model does not have OA specified as per person, then we need to report and error. In this case I believe it should terminate the program, since the user can then correct the problem and re-run.